logic/strategy/card_utils.go

```diff
@@ -36,41 +36,28 @@ func otherOptions(desired int, avoid map[model.Card]struct{}) [][]model.Card {
 	return options
 }
 
-func chooseFrom(desired int, hand []model.Card) [][]model.Card { //nolint:gocyclo
-	if desired > len(hand) || desired > 4 || desired <= 0 {
-		return nil
-	}
-	// 6 choose 2 = 15, our largest number of potential deposits to the crib
-	allDeposits := make([][]model.Card, 0, 15)
-	for i, c1 := range hand {
-		if desired == 1 {
-			// it's a three or four player game, return one card
-			allDeposits = append(allDeposits, []model.Card{c1})
-			continue
+func chooseFrom(k int, hand []model.Card) [][]model.Card {
+	if k == 1 {
+		all := make([][]model.Card, len(hand))
+		for i, e := range hand {
+			all[i] = []model.Card{e}
 		}
-		for j := i + 1; j < len(hand); j++ {
-			c2 := hand[j]
-			if desired == 2 {
-				allDeposits = append(allDeposits, []model.Card{c1, c2})
-				continue
-			}
-			for k := i + 1; k < len(hand); k++ {
-				c3 := hand[k]
-				if desired == 3 {
-					allDeposits = append(allDeposits, []model.Card{c1, c2, c3})
-					continue
-				}
-				for l := i + 1; l < len(hand); l++ {
-					c4 := hand[l]
-					if desired == 4 {
-						allDeposits = append(allDeposits, []model.Card{c1, c2, c3, c4})
-						continue
-					}
-				}
-			}
+		return all
+	}
+	if k == len(hand) {
+		return [][]model.Card{hand}
```

Comment:
Could we copy `hand` here before returning it? Just to avoid any weird memory issues.

```go
cpy := make([]model.Card, len(hand))
copy(cpy, hand)
return [][]model.Card{cpy}
```